### PR TITLE
agendadb: Fix unhandled errors while fetching all agendas

### PR DESCRIPTION
**Pull request by dmigwi**
When fetching all agendas if opening the db failed, the application still proceeded to use an invalid agendas db connection failed.

- [x] Fixed unhandled errors

https://github.com/decred/dcrdata/issues/587

### CODE REVIEW

**Comment by dmigwi** [comment]
@chappjc Please have a look at this when you get the time.

File: db/agendadb/db.go [hunk 1]
```diff
@@ -220,13 +220,20 @@ func GetAllAgendas() (agendas []*AgendaTagged, err error) {
 	adb, err := Open(dbName)
 	if err != nil {
 		log.Errorf("Failed to open new DB: %v", err)
+		return
 	}
+
+	defer func() {
+		err = adb.Close()
+		if err != nil {
+			log.Errorf("Failed to close the DB: %v", err)
```
Comment (chappjc) [inline db/agendadb/db.go]: Please mention agenda DB, otherwise it could be confusing what threw the error.

File: db/agendadb/db.go [hunk 2]
```diff
@@ -220,13 +220,20 @@ func GetAllAgendas() (agendas []*AgendaTagged, err error) {
 	adb, err := Open(dbName)
```
Comment (chappjc) [inline db/agendadb/db.go]: I believe err here shadows the err declared in the function signature

File: db/agendadb/db.go (hunk 1 above)
Comment (dmigwi) [reply db/agendadb/db.go]: Ok. Noted

File: db/agendadb/db.go (hunk 2 above)
Comment (dmigwi) [reply db/agendadb/db.go]: Ok. Noted

**Pull request merged**